### PR TITLE
Added overflowGalleryLayout prop to VideoGallery

### PR DESCRIPTION
**Pull request by mgamis-msft**
# What

- Added overflowGalleryLayout prop to VideoGallery to control which component to use for overflow participants
- Renamed VideoGalleryResponsiveHorizontalGallery -> OverflowGallery
- Refactor FloatingLocalVideo component to simply have localVideoSize prop instead of using isNarrow boolean prop to decide the size internally

# Why

https://skype.visualstudio.com/SPOOL/_workitems/edit/3151770

# How Tested

Local storybook preview testing. Relying on snapshot tests on CI.

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-components/src/components/VideoGallery/styles/FloatingLocalVideo.styles.ts [hunk 1]
```diff
@@ -38,6 +38,12 @@ export const SMALL_FLOATING_MODAL_SIZE_PX = { width: 58, height: 104 };
  */
 export const LARGE_FLOATING_MODAL_SIZE_PX = { width: 215, height: 120 };
 
+/**
+ * Vertical gallery floating modal width and height in rem
+ * Aspect ratio: 16:9
+ */
+export const VERTICAL_GALLERY_FLOATING_MODAL_SIZE_PX = { width: 144, height: 81 };
```
Comment (dmceachernmsft) [inline packages/react-components/src/components/VideoGallery/styles/FloatingLocalVideo.styles.ts:45]: should this be 144 x 90? I guess thats not 16: 9 hey?

File: packages/react-components/src/components/VideoGallery/styles/VideoGalleryResponsiveVerticalGallery.styles.ts [hunk 2]
```diff
@@ -13,24 +14,36 @@ import { LARGE_FLOATING_MODAL_SIZE_PX, SMALL_FLOATING_MODAL_SIZE_PX } from './Fl
...
 };
+
+/**
+ * @private
+ */
+export const VERTICAL_GALLERY_TILE_STYLE = {
+  minHeight: `${VERTICAL_GALLERY_TILE_SIZE_REM.minHeight}rem`,
+  minWidth: `${VERTICAL_GALLERY_TILE_SIZE_REM.width}rem`,
+  maxHeight: `${VERTICAL_GALLERY_TILE_SIZE_REM.maxHeight}rem`,
+  maxWidth: `${VERTICAL_GALLERY_TILE_SIZE_REM.width}rem`
```
Comment (dmceachernmsft) [inline packages/react-components/src/components/VideoGallery/styles/VideoGalleryResponsiveVerticalGallery.styles.ts:42]: I think we can just define this as width and we don't need a min and max here

File: packages/react-components/src/components/VideoGallery.tsx [hunk 3]
```diff
@@ -276,7 +290,9 @@ export const VideoGallery = (props: VideoGalleryProps): JSX.Element => {
     /* @conditional-compile-remove(pinned-participants) */
     onUnpinParticipant: onUnpinParticipantHandler,
     /* @conditional-compile-remove(pinned-participants) */
-    remoteVideoTileMenuOptions = DEFAULT_REMOTE_VIDEO_TILE_MENU_OPTIONS
+    remoteVideoTileMenuOptions = DEFAULT_REMOTE_VIDEO_TILE_MENU_OPTIONS,
+    /* @conditional-compile-remove(vertical-gallery) */
+    overflowGalleryLayout = 'VerticalRight'
```
[the remark this answers is not in the document]
Comment (mgamis-msft) [reply packages/react-components/src/components/VideoGallery.tsx:295]: This should be 'HorizontalBottom'. Good catch!

File: packages/react-components/src/components/VideoGallery/styles/FloatingLocalVideo.styles.ts (hunk 1 above)
Comment (mgamis-msft) [reply packages/react-components/src/components/VideoGallery/styles/FloatingLocalVideo.styles.ts:45]: Oh I just followed the size in Figma and it is the following currently:
image

File: packages/react-components/src/components/VideoGallery/styles/VideoGalleryResponsiveVerticalGallery.styles.ts (hunk 2 above)
Comment (mgamis-msft) [reply packages/react-components/src/components/VideoGallery/styles/VideoGalleryResponsiveVerticalGallery.styles.ts:42]: I remember there was an issue with that simplification from the past but let me test if I can make that simplification.

**Review by dmceachernmsft — APPROVED** [review]

File: packages/communication-react/review/beta/communication-react.api.md [hunk 4]
```diff
@@ -3237,6 +3240,7 @@ export interface VideoGalleryProps {
     onRenderLocalVideoTile?: (localParticipant: VideoGalleryLocalParticipant) => JSX.Element;
     onRenderRemoteVideoTile?: (remoteParticipant: VideoGalleryRemoteParticipant) => JSX.Element;
     onUnpinParticipant?: (userId: string) => void;
+    overflowGalleryLayout?: OverflowGalleryLayout;
```
Comment (JamesBurnside) [inline packages/communication-react/review/beta/communication-react.api.md:3243]: nit this is already a VideoGallery so `Gallery` isn't needed in the naming here, to match the existing `layout` props should this instead be `overflowLayout: OverflowGalleryLayout`?

**Review by JamesBurnside — APPROVED** [review]

File: packages/communication-react/review/beta/communication-react.api.md (hunk 4 above)
Comment (mgamis-msft) [reply packages/communication-react/review/beta/communication-react.api.md:3243]: Really good point here. This new prop will pertain to how VideoGallery will layout overflow participants and overflowLayout would be a concise name. But we might want to keep the naming connection of inner overflow components HorizontalGallery, VerticalGallery etc to this new prop so overflowGalleryLayout maintains a stronger connections. We can always update it in beta if we start getting signals we are exporting the components if ARB picks on it.

**Pull request merged**